ingestion/src/lib.rs

```diff
@@ -425,6 +433,32 @@ fn get_conversions(filename: &str) -> Result<ParamConversions, csv::Error> {
     ))
 }
 
+/// Middleware function that runs around a request, so we can record how long it took
+async fn track_request_duration(req: Request, next: Next) -> impl IntoResponse {
+    let start = std::time::Instant::now();
+    let path = if let Some(matched_path) = req.extensions().get::<MatchedPath>() {
+        matched_path.as_str().to_owned()
+    } else {
+        req.uri().path().to_owned()
+    };
+    let method = req.method().clone();
+
+    let response = next.run(req).await;
+
+    let latency = start.elapsed().as_secs_f64();
```

Comment:
I suggest to rename _latency_ to e.g. _duration_ to reflect that the value includes the time spent processing the request rather than just the time from the initiation to onset.

Reply:
Done, my rationale for not using "latency" (as I did in the function and metric name) is a little different though. I interpret latency here to mean time from the client sending the request to receiving a response (i.e duration + ping) which we can't measure on the server side